### PR TITLE
Fix for issue #18526

### PR DESCRIPTION
**Pull request by darkdef**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ✔️/❌
| Tests pass?   | ✔️
| Fixed issues  | #18526 
Fix for: DbCache doesn't working using MSSQL/SqlServer

### CODE REVIEW

File: framework/caching/DbCache.php [hunk 1]
```diff
@@ -58,12 +58,20 @@ class DbCache extends Cache
...
+     *     data VARBINARY(MAX)
+     * );
+     * ```
+     *
      * where 'BLOB' refers to the BLOB-type of your preferred DBMS. Below are the BLOB type
      * that can be used for some popular DBMS:
      *
      * - MySQL: LONGBLOB
      * - PostgreSQL: BYTEA
-     * - MSSQL: BLOB
```
Comment (samdark) [inline framework/caching/DbCache.php:66]: - MSSQL: VARBINARY(MAX)

**Review by samdark — APPROVED** [review]
Looks very hacky but don't see a better way :(

File: framework/caching/DbCache.php [hunk 2]
```diff
@@ -286,4 +294,36 @@ protected function flushValues()
...
+    /**
+     * @return bool
+     *
+     * Checking Mssql: if field is varbinary - return true
+     */
+    protected function isVarbinaryDataField()
+    {
+        return (
+            in_array($this->db->getDriverName(), ['sqlsrv', 'dblib']) &&
+            $this->db->getTableSchema('cache')->columns['data']->dbType === 'varbinary');
```
Comment (bizley) [inline framework/caching/DbCache.php:307]:
```suggestion
        return in_array($this->db->getDriverName(), ['sqlsrv', 'dblib'])
            && $this->db->getTableSchema('cache')->columns['data']->dbType === 'varbinary';
```

Comment (bizley) [reply framework/caching/DbCache.php:307]: Also please store the result in a private property. This method is used several times here and while some parts of the conditions are inner-cached already there are the ones that are recalculated every time.

**Review by bizley — CHANGES_REQUESTED** [review]

File: framework/caching/DbCache.php (hunk 2 above)
Comment (darkdef) [reply framework/caching/DbCache.php]: Resolved

**Review by bizley — APPROVED** [review]
LGTM. Please add changelog.

**Comment by bizley** [comment]
👍🏻

**Pull request merged**